crates/common/src/provider/retry.rs

```diff
@@ -84,6 +84,10 @@ fn should_retry_json_rpc_error(error: &ErrorPayload) -> bool {
         return true
     }
 
+    if *code == -32005 && message.contains("archive request rate exceeded") {
```

Comment:
Can you add a comment for it?

Reply:
actually, this is weird. it should have been caught by this already:

https://github.com/foundry-rs/foundry/blob/master/crates/common/src/provider/retry.rs#L77-L78

which makes me think that retryprovider is not properly configured or it's not properly handling the errors:

https://github.com/foundry-rs/foundry/blob/master/crates/common/src/provider/retry.rs#L31-L44